src/test/java/com/segmentreader/mapreduce/AbstractCookieReducerTestCase.java

```diff
@@ -33,23 +40,29 @@ private AbstractCookieReducer createInstance (Map<String, OperationHandler> hand
     @Test
     public void testAbstractCookieReducer() throws IOException, InterruptedException {
         OperationHandler handler = mock(OperationHandler.class);
+        ArgumentCaptor<UserModCommand> umcCaptor = ArgumentCaptor.forClass(UserModCommand.class);
         Counter mapRedCounter = mock(Counter.class);
+        Instant timeStamp = Instant.EPOCH;
         Map<String, OperationHandler> handlers = ImmutableMap.of(OperationHandler.DELETE_OPERATION, handler,
                 OperationHandler.ADD_OPERATION, handler);
-        UserModCommand umc = new UserModCommand("11",OperationHandler.ADD_OPERATION, 
-                Arrays.asList("iphone", "macbook", "magic mouse"));
-        UserModCommand umc1 = new UserModCommand("11",OperationHandler.DELETE_OPERATION, 
+        List<String> umc1Segments = Arrays.asList("iphone", "macbook", "magic mouse");
+        Collections.reverse(umc1Segments);
+        List<String> umc2Segments = Arrays.asList("dakine bag", "dakine case", "dakine gloves");
+        Collections.reverse(umc2Segments);
+        UserModCommand umc = new UserModCommand(timeStamp,"11",OperationHandler.ADD_OPERATION,umc1Segments);
+        UserModCommand umc1 = new UserModCommand(timeStamp,"11",OperationHandler.DELETE_OPERATION,
                 Arrays.asList("dakine bag", "dakine case", "dakine gloves"));
-
         Iterable<UserModCommand> values = Arrays.asList(umc,umc1);
         Context context = mock(Context.class);
         
         when(context.getCounter("segmentreader", "reduce_counter")).thenReturn(mapRedCounter);
         
         AbstractCookieReducer reducer = createInstance(handlers);
-        reducer.reduce(new Text("1"), values, context);
+        reducer.reduce(new Text("11"), values, context);
  
-        verify(handler, times(2)).handle(any(UserModCommand.class));
+        verify(handler, times(2)).handle(umcCaptor.capture());
         verify(mapRedCounter, times(1)).increment(1);
+        List<UserModCommand> expected = Lists.newArrayList(values);
+        assertThat(expected, is(umcCaptor.getAllValues()));
     }
```

Comment:
Also I asked you before to move some expected defenitions to corresponding section. ?
What do you compare here? stritly speaking, it is not fair to compare the same objects. Why do not you compare some aggragations, you have made with your streams?